packages/waku/src/lib/vite-entries/entry.ssr.tsx

```diff
@@ -1,14 +1,15 @@
 import renderHtmlEnhancer from 'virtual:vite-rsc-waku/render-html-enhancer';
-import type { Unstable_RenderHtml as RenderHtml } from '../types.js';
 
 export {
   renderHtmlStream as INTERNAL_renderHtmlStream,
   renderHtmlFallback as INTERNAL_renderHtmlFallback,
 } from '../vite-rsc/ssr.js';
 
-export async function INTERNAL_enhanceRenderHtml(renderHtml: RenderHtml) {
+export async function INTERNAL_enhanceRenderHtml(
+  render: () => Promise<Response>,
+): Promise<Response> {
   if (renderHtmlEnhancer) {
-    renderHtml = renderHtmlEnhancer(renderHtml);
+    return renderHtmlEnhancer(render);
   }
-  return renderHtml;
+  return render();
```

Comment:
I don't completely dislike this idea. This is limited, but simpler.
It's just naming what's weird. As it's no longer enhancing, something like this would be good:
```js
export async function INTERNAL_transformRenderHtmlResponse(
  res: Response
): Promise<Response> {
  // ...
}
```